File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;
import entity.PageResult;

/**
 * 品牌接口
 * @author 13191
 *
 */
public interface BrandService {


	public List<TbBrand> findAll();

	/**
	 * 品牌管理
	 * @param pagNum 当前页面
	 * @param pageSize 每页记录数
	 * @return
	 */
	public PageResult findPage(int pagNum,int pageSize);

	/**
	 * 增加
	 * @param brand
	 */
	public void add(TbBrand brand);

	/**
	 * 根据ID查询实体
	 * @param id
	 * @return
	 */
	public TbBrand findOne(Long id);

	/**
	 * 修改
	 * @param brand
	 */
	public void update(TbBrand brand);

	/**
	 * 删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 搜索分页
	 * @param brand
	 * @param pagNum
	 * @param pageSize
	 * @return
	 */
	public PageResult findPage(TbBrand brand,int pagNum,int pageSize);

	/**
	 * 返回下拉列表数据
	 * @return
	 */
	public List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
//品牌服务
app.service('brandService', function ($http) {
    //读取列表数据绑定到表单
    this.findAll=function () {
        return $http.get('../brand/findAll.do');
    }

    //分页
    this.findPage=function (page, size) {
        return $http.get('../brand/findPage.do?page=' + page + '&size=' + size);
    }

    //增加
    this.add=function (entity) {
        return $http.post('../brand/add.do', entity);
    }

    //修改
    this.update=function (entity) {
        return $http.post('../brand/update.do', entity);
    }

    //findOne
    this.findOne=function (id) {
        return $http.get('../brand/findOne.do?id=' + id);

    }

    //删除
    this.dele=function (selectIds) {
        return $http.get('../brand/delete.do?ids=' + selectIds);
    }

    //搜索
    this.search=function (page, size,searchEntity) {
        return $http.post('../brand/search.do?page=' + page + '&size=' + size,searchEntity)
    }

    //下拉列表数据
    this.selectOptionList=function () {
        return $http.get('../brand/selectOptionList.do');
    }
});